### ISSUE TITLE
Create an example for fullscreen mode

### ISSUE THREAD

**Issue by lucaswoj** [mapbox/mapbox-gl-js#1824]
mapbox.js has [such an example](https://www.mapbox.com/mapbox.js/example/v1.0.0/leaflet-fullscreen)

ref https://github.com/mapbox/mapbox-gl-js/issues/1821#issuecomment-163470008

cc @jingsam

**Comment by josiekre** [comment]
+1

**Comment by dshakya** [comment]
Is the team working on implementing Fullscreen plugin for MapboxGL? Missing this feature big time.

**Comment by lucaswoj** [comment]
@dshakya While there is no pre-built plugin, you should be able to implement this feature downstream:

```js
map.getCanvas().webkitRequestFullScreen(); // there are a few other vendor prefixes to try
map.resize();
```

**Comment by colleenmcginnis** [comment]
@lucaswoj I was thinking of working on this example. Do you think we should build a control or just show users how to implement what you suggested above?

**Comment by dshakya** [comment]
Would be lovely if you could build a control like the one for Leaflet

**Comment by dshakya** [comment]
Hi @colleenmcginnis, would you know if anyone is working on this feature? Or any directions on how would I go about implementing it?

**Comment by karlguillotte** [comment]
@lucaswoj I created that fullscreen control. Would you be interested in a PR?

**Comment by andrewharvey** [comment]
@karlguillotte Not sure if you saw [this one by @colleenmcginnis](https://github.com/mapbox/mapbox-gl-js/commit/4c5aff4f3319c15886ed72d370176c634b822913)  to compare?

**Comment by karlguillotte** [comment]
Thanks @andrewharvey ! I have not seen that one yet.

**Comment by shotor** [comment]
I tried this, but my controls disappear in fullscreen mode.

Doesn't work on JsFiddle, so here's a gist: https://gist.github.com/shotor/316d2dc8b07fb69b6b73e4589b07b6ce

Has anyone been able to get the controls to display in full screen?

**Comment by anx-ckreuzberger** [comment]
Any chance the fullscreen stuff from [this commit](https://github.com/mapbox/mapbox-gl-js/commit/4c5aff4f3319c15886ed72d370176c634b822913) can make it into the master? Is there a PR or something?
Pinging @colleenmcginnis on this too

**Comment by mollymerp** [comment]
this has shipped in #3977 !! will be available in the next release!
